### PR TITLE
EN-2400-datatable-task-list-claim-button

### PR DESCRIPTION
**Pull request by alepintus**
Hi @joewhite101 ,

this is the cherry-pick from [this PR](https://github.com/entando/entando-components/pull/478).
Should be  be merged after these PRs:
https://github.com/entando/entando-components/pull/486
https://github.com/entando/entando-components/pull/487

Thanks!

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceCommentsAction.java [hunk 1]
```diff
@@ -34,18 +41,117 @@
...
+                this.setFrontEndCaseData(frontEndCaseDataIn);
+                String channelIn = extractWidgetConfig("channel");
+                this.setChannel(channelIn);
+
+                this.setKnowledgeSourceId(frontEndCaseDataInjs.getString("knowledge-source-id"));
+                this.setContainerid(frontEndCaseDataInjs.getString("container-id"));
+                this.setChannelPath(this.getChannel());
+
+                config = formManager.getKieServerConfigurations().get(frontEndCaseDataInjs.getString("knowledge-source-id"));
+//                List<String> cases = this.caseManager.getCaseInstancesList(config, this.getContainerid());
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/internalservlet/BpmCaseInstanceCommentsAction.java]: Remove if dead.

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java [hunk 2]
```diff
@@ -593,6 +629,35 @@ public void putTaskDoc(final KiaApiTaskDoc doc) throws Throwable {
...
+                if (StringUtils.isNotEmpty(information)) {
+                    final ApsProperties config = new ApsProperties();
+                    config.loadFromXml(information);
+                    String knowledgetSource = (String) config.get(KieBpmSystemConstants.WIDGET_INFO_PROP_KIE_SOURCE_ID);
+                    KieBpmConfig bpmConfig = this.getKieFormManager().getKieServerConfigurations().get(knowledgetSource);
+                    String containerId = config.getProperty("containerId");
+                    String taskId = claimTask.getTaskId();
+                    
+                    //TODO Check the user passed to the API
+                    String username = bpmConfig.getUsername();
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java]: @alepintus We need to update this to use the logged in user if there is one. Same change for 5.0.3

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/KieFormManager.java [hunk 3]
```diff
@@ -1036,6 +1041,13 @@ public KieProcessInstancesQueryResult getProcessInstancesWithClientData(KieBpmCo
         return result;
     }
 
+//    public ConfigInterface getConfigManager() {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/KieFormManager.java]: Remove if dead

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/KieFormManager.java [hunk 4]
```diff
@@ -1044,6 +1056,24 @@ public void setOverrideManager(IKieFormOverrideManager overrideManager) {
         this.overrideManager = overrideManager;
     }
 
+//    @Override
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/kie/KieFormManager.java]: Remove if dead

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java (hunk 2 above)
Comment (alepintus) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java]: @joewhite101 Can I create a new task on Jira for this change?

Comment (joewhite101) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java]: @alepintus Yeah. We can merge this without that change as long as we get a Jira opened.

Comment (alepintus) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/api/ApiTaskInterface.java]: @joewhite101 new task created

**Comment by entando-service** [comment]
All is Well

**Pull request merged**